egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Pegah Ghahremani
+#
+# Apache 2.0.
+
+""" This script generates egs.archive.scp and ranges.* used for multilingual setup.
+    Also this script generates outputs.*.ark and weight.*.ark, where each line
+    corresponds to language-id and weight for the same example in egs.*.scp.
+    weight.*.ark is archive of table indexed by the key of input example
```

Comment:
each line "i s1 n1" in ranges.*.txt corresponds to  selecting "n1" examples starting from line "s1" from i_th scp-file in egs_scp_lists.